File: 2094-finding-3-digit-even-numbers/2094-finding-3-digit-even-numbers.cpp
class Solution {
public:
    vector<int> findEvenNumbers(vector<int>& digits) {
        int freq[10] ={0}; //store freq of each digit
        vector<int> result;
        
        for(int i: digits)
        {
            freq[i]++;
        }
        
        for(int num=100; num<999; num+=2)
        {
            int ones=num%10;
            int tens=(num/10)%10;
            int hundreds=(num/100);
            
            freq[ones]--;
            freq[tens]--;
            freq[hundreds]--;
            
            if(freq[ones]>=0 && freq[tens]>=0 && freq[hundreds]>=0)
            {
                result.push_back(num);
            }
            freq[ones]++;
            freq[tens]++;
            freq[hundreds]++;
            
        }
        
        return result;
    }
};

File: 2094-finding-3-digit-even-numbers/2094-finding-3-digit-even-numbers.java
class Solution {
    public int[] findEvenNumbers(int[] digits) {
        TreeSet<Integer> set= new TreeSet<>();
        
        for(int i=0; i<digits.length; i++)
        {
            if(digits[i]==0)
            {
                continue;
            }
            for(int j=0; j<digits.length;j++)
            {
                if(i==j)
                {
                    continue;
                }
            
                for(int k=0; k<digits.length; k++)
                {
                    if(i==k || j==k || digits[k]%2==1)
                    {
                        continue;
                    }
                    int val=100*digits[i]+10*digits[j]+digits[k];
                    set.add(val);
                }
            }
        }
        int[] result=new int[set.size()];
        int i=0;
        
        for(int j: set)
        {
            result[i]=j;
            i++;
        }
        return result;
    }
}